dogstatsd.mli

```diff
@@ -0,0 +1,140 @@
+module Base : sig
+  module Tag : sig
+    type t = string * string (** (tag-name, value) *)
+  end
+
+  (** https://docs.datadoghq.com/developers/dogstatsd/#metrics *)
+  module Metric : sig
+    module Counter : sig
+      type t =
+        [ `Increment
+        | `Decrement
+        | `Value of int ]
+    end
+
+    type typ =
+      [ `Counter of Counter.t
+      | `Gauge of float
+      | `Timer of float
+      | `Histogram of int
+      | `Set of string ]
+
+    type t =
+      { metric_name : string
+      ; metric : typ
+      ; sample_rate : float option
+      ; tags : Tag.t list }
+  end
+
+  (** https://docs.datadoghq.com/developers/dogstatsd/#service-checks *)
+  module ServiceCheck : sig
+    type status =
+      [ `Ok
+      | `Warning
+      | `Critical
+      | `Unknown ]
+
+    type t =
+      { name : string
+      ; status : status
+      ; timestamp : int option
+      ; hostname : string option
+      ; tags : Tag.t list
+      ; message: string option }
+  end
+
+  (** https://docs.datadoghq.com/developers/dogstatsd/#events *)
+  module Event : sig
+    type priority = [ `Normal | `Low ]
+
+    type alert =
+      [ `Error
+      | `Warning
+      | `Info
+      | `Success ]
+
+    type t =
+      { title : string
+      ; text : string
+      ; timestamp : int option
+      ; hostname : string option
+      ; aggregation_key : string option
+      ; priority : priority option (* defaults to Normal *)
+      ; source_type_name : string option
+      ; alert_type : alert option (* defaults to Info *)
+      ; tags: Tag.t list }
+  end
+end
+
+(* TODO: replace IO with {Async, Lwt, Sync} *)
```

Comment:
This is the intention of this comment. I've done something similar in the separate packages pull request. I'm still figuring out how it will all work together wrt this extended interface.